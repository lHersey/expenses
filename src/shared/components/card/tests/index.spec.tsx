import { render } from '@testing-library/react-native';
import React from 'react';
import Text from 'shared/components/text';
import Providers from 'shared/providers/test-providers';

import Card from '../index';

describe('<Card />', () => {
  describe('snapshots', () => {
    it('should match default snapshot', () => {
      expect.hasAssertions();
      const rendered = render(
        <Providers>
          <Card>
            <Text>Hello Card</Text>
          </Card>
        </Providers>,
      ).toJSON();

      expect(rendered).toMatchSnapshot();
    });

    it('should match gradient snapshot', () => {
      expect.hasAssertions();
      const rendered = render(
        <Providers>
          <Card gradient>
            <Text>Hello Card</Text>
          </Card>
        </Providers>,
      ).toJSON();

      expect(rendered).toMatchSnapshot();
    });
  });

  describe('unit', () => {
    it('should render the text', () => {
      expect.hasAssertions();
      const rendererd = render(
        <Providers>
          <Card>
            <Text>Hello Card</Text>
          </Card>
        </Providers>,
      );

      expect(rendererd.queryByText('Hello Card')).toBeTruthy();
    });
  });
});
