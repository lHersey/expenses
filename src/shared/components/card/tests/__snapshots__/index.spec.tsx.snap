// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> snapshots should match default snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "elevation": 8,
          "overflow": "visible",
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 4,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#FFF",
            "borderRadius": 15,
            "display": "flex",
            "flexGrow": 1,
            "overflow": "hidden",
            "paddingBottom": 15,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 15,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#3e3e3e",
              "fontSize": 16,
            },
          ]
        }
      >
        Hello Card
      </Text>
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`<Card /> snapshots should match gradient snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "elevation": 8,
          "overflow": "visible",
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 4,
        },
      ]
    }
  >
    <ViewManagerAdapter_ExpoLinearGradient
      proxiedProperties={
        Object {
          "colors": Array [
            4280987875,
            4292112360,
            4294874740,
          ],
          "endPoint": Array [
            1,
            0,
          ],
          "locations": undefined,
          "startPoint": Array [
            0,
            0,
          ],
        }
      }
      style={
        Array [
          Object {
            "backgroundColor": "#FFF",
            "borderRadius": 15,
            "display": "flex",
            "flexGrow": 1,
            "overflow": "hidden",
            "paddingBottom": 15,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 15,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#3e3e3e",
              "fontSize": 16,
            },
          ]
        }
      >
        Hello Card
      </Text>
    </ViewManagerAdapter_ExpoLinearGradient>
  </View>
</RNCSafeAreaProvider>
`;
